native/src/snappyclient/cpp/impl/ControlConnection.cpp

```diff
@@ -301,6 +301,11 @@ void ControlConnection::failoverToAvailableHost(
             || m_snappyServerType
                 == thrift::ServerType::THRIFT_SNAPPY_CP_SSL) {
           TSSLSocketFactory sslSocketFactory;
+          std::string sslProperty = ClientService::getSSLPropertyName(SSLProperty::TRUSTSTORE);
+          std::string trustStoreCert;
+          ClientService::getSSLPropertyValue(sslProperty, trustStoreCert);
+          sslSocketFactory.loadTrustedCertificates(trustStoreCert.c_str());
+          sslSocketFactory.authenticate(false);
           tTransport = sslSocketFactory.createSocket(controlAddr.hostName,
               controlAddr.port);
```

Comment:
This code can be moved in a common static method in ClientService that will return the result of createSocket.

Reply:
Moved into the method name ClientService::createSocket